### PR TITLE
Sets up Data Browser

### PR DESCRIPTION
**Pull request by GabePires**

### CODE REVIEW

**Review by GabePires — COMMENTED** [review]
1. Created ChallengeResult and UserResult. These are just codables to help us parse the sample json data (challenges.json and users.json). To see the parsing and assigning of data, see DataController: populateWithSampleData()

2. Initialize PhotoController in iSpyTableViewController and pass in if it is "PhotoInjectable"

3. Added protocol for User injectable and Challenge injectable and Match injectable

4. Added new tables and storyboards for the flow

5. Changed longitude in Challenge managed object from type String to Double

6. Only populate with sample data once based on whether or not the persistent store was set up already (previous challenge was like this as well).

File: iSpyChallenge/Data Browser/ChallengesTableViewController.swift [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+class ChallengesTableViewController: UITableViewController, DataControllerInjectable, PhotoControllerInjectable, UserInjectable {
+    
+    var dataController: DataController!
+    var photoController: PhotoController!
+    var user: User?
+    
+    private lazy var fetchedResultsController: NSFetchedResultsController<Challenge> = {
+        let fetchRequest: NSFetchRequest<Challenge> = Challenge.newFetchRequest()
+        fetchRequest.predicate = NSPredicate(format: "creator = %@", self.user!)
```
Comment (quicksort) [inline iSpyChallenge/Data Browser/ChallengesTableViewController.swift:18]: Given that we are force-unwrapping the user property anyway (i.e. given that it is necessary to have it), should we define the protocol as "var user: User! { get set }"?

Comment (GabePires) [reply iSpyChallenge/Data Browser/ChallengesTableViewController.swift:18]: Good question. Originally, I had it like that. There is one view controller, the MatchTableViewController, where the user variable and challenge variable are both optional. In order to make all the "UserInjectables" and "ChallengeInjectables" happy, it had to be optional.

File: iSpyChallenge/Data Browser/MatchesTableViewController.swift [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            
+        @unknown default:
+            break
+        }
+    }
+    
+    func controllerDidChangeContent(_ controller: NSFetchedResultsController<NSFetchRequestResult>) {
+        tableView.endUpdates()
+    }
+}
```
Comment (quicksort) [inline iSpyChallenge/Data Browser/MatchesTableViewController.swift:132]: Given how many times we use it, perhaps we could abstract it to a protocol with a protocol implementation. This is just a code challenge, however, and so perhaps it doesn't matter (we could even ask the candidate how he'd remove the duplication).

**Review by quicksort — APPROVED** [review]

File: iSpyChallenge/Data Browser/MatchesTableViewController.swift (hunk 2 above)
Comment (GabePires) [reply iSpyChallenge/Data Browser/MatchesTableViewController.swift:132]: Good point, that would be nice. Trying that (protocol + protocol implementation) out locally, however, I get some odd warnings/errors regarding @objc and @nonobjc.

**Review by quicksort — APPROVED** [review]

**Pull request merged**